docs/examples/Naive_Classification.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+"""
+Forward Algorithm
+=================
+An example of the forward algorithm using Stone Soup components.
+"""
+
+# %%
+# Modelling the state space
+# -------------------------
+# We model a target that can be one of three categories: Bicycle, Car or Bus.
+# In this simple instance, the target does not switch between these categories (ie. if it is a bus,
+# it stays a bus).
+# This is modelled in the state space by 3 state vector elements. The first component indicates
+# the probability that the target is a bicycle, the second that it is a car, and the third that it
+# is a bus.
+# For this model, it will remain a bus, therefore the 3rd state vector component will be
+# :math:`1` throughout, with the other components remaining :math:`0`.
+
+import numpy as np
+from stonesoup.types.state import State, StateVector
+
+np.random.seed(1991)
+
+gt = []
+nsteps=100
+
+for step in range(nsteps):
+    gt.append(State(StateVector([0, 0, 1])))  # bicycle, car, bus
+
+
+# %%
+# Modelling state transition
+# --------------------------
+# We require a model of the state transition in order to track the target's classification.
+# This will be a simple linear map, represented by a :math:`3\times 3` matrix with no noise added.
+# We might not know that the target does not switch classification, so the matrix should not simply
+# be the identity matrix.
+#
+# .. math::
+#       T = \begin{bmatrix}
+#  		P(Bicycle_{t}|Bicycle_{t-1}) & P(Bicycle_{t}|Car_{t-1}) & P(Bicycle_{t}|Bus_{t-1}) \\
+#  		P(Car_{t}|Bicycle_{t-1}) & P(Car_{t}|Car_{t-1}) & P(Car_{t}|Bus_{t-1}) \\
+#  		P(Bus_{t}|Bicycle_{t-1}) & P(Bus_{t}|Car_{t-1}) & P(Bus_{t}|Bus_{t-1})
+#  	    \end{bmatrix}
+#
+# The transitioned state is given by
+# :math:`f(X_{t-1})_{j} = T_{ij}{X_{t-1}}_{i} = T^T_{ji}{X_{t-1}}_{i}`.
+# Therefore, we take the transpose :math:`f(X_{t-1}) = T^{T}X_{t-1}` when transitioning the state.
+
+F = np.array([[0.8, 0.1, 0.1],
+              [0.1, 0.8, 0.1],
+              [0, 0.1, 0.9]])
+
+
+def transit(state):
+    return F.T @ state.state_vector
+
+
+# %%
+# Modelling observations
+# ----------------------
```

Comment:
I think it's important to state that we don't observe the state directly (hence 'hidden'). We observe 'emissions' governed by a matrix of probabilities `E`. (We could add noise to this but again, bit more complicated than additive.)